rust%intel doesn't build

### Steps to reproduce

```console
spack install rust%intel
```

### Error message

```console
==> Installing rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh [14/14]
==> No binary for rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh found: installing from source
==> Warning: Intel's compilers may or may not optimize to the same degree for non-Intel microprocessors for optimizations that are not unique to Intel microprocessors
==> Using cached archive: /n/spack_source_cache/_source-cache/archive/96/96d62e6d1f2d21df7ac8acb3b9882411f9e7c7036173f7f2ede9e1f1f6b1bb3a.tar.gz
==> No patches needed for rust
==> rust: Executing phase: 'configure'
==> rust: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/python3.6' './x.py' 'build'

5 errors found in build log:
     78       Compiling ignore v0.4.18
     79       Compiling fd-lock v3.0.11
     80       Compiling clap_derive v4.2.0
     81       Compiling clap v4.2.4
     82       Compiling clap_complete v4.2.2
     83       Compiling build_helper v0.1.0 (/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/src/tools/build_helper)
  >> 84    error: linking with `cc` failed: exit status: 1
     85      |
     86      = note: LC_ALL="C" PATH="/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-
           linux-gnu/bin:/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/x86_64-unknown-linux-gnu/stage0/bin:/opt/spack_nov23/var/spack/environments/intel/install
           /linux-centos8-zen3/intel-2021.10.0/libgit2-1.7.0-rrwks2nc6dc6dvzwhu7qmm6nbixlfifl/bin:/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/curl-8.4.0-q24pbykkelo
           tqprhwm5dqr77pppi27nc/bin:/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/openssl-3.1.3-ktfw4fp6waeos6gqwhex3nogrrjtlo23/bin:/opt/spack_nov23/var/spack/envir
           onments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/bin:/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-202
           1.10.0/pkgconf-1.9.5-yx6ge5zhgnp6ysnfo3tg2pc3kdivcn6w/bin:/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/ninja-1.11.1-dofknaax44vaxldcrtvv4dnvcziu4uo7/bin:/
           opt/spack_nov23/lib/spack/env/intel:/opt/spack_nov23/lib/spack/env/case-insensitive:/opt/spack_nov23/lib/spack/env:/opt/spack_nov23/bin:/opt/xxxxxx/bin/:/home/user/bin:/opt/xxxxxx/bin/:/home/user/bi
           n:/opt/xxxxxx/bin/:/usr/share/Modules/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin" VSLANG="1033" "cc" "-m64" "/tmp/rustctUYEHB/symbols.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loi
           ur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.164xh7gax70rersm.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh
           /spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.1707x4p4vxoa6iob.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/de
           bug/deps/bootstrap-8d477eb8daba6006.1g3eg6vjlev89hjh.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8da
           ba6006.1l4qsni41jhnzbea.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.1o37kkiqrslr28yz.rcgu.
           o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.208uo7kef44bgk9k.rcgu.o" "/tmp/user/spack-stage/spa
           ck-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.2220bxgdy439duav.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ft
           bc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.2na1qogg2g1nri8g.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spac
           k-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.2swrrnneurhydcas.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/d
           eps/bootstrap-8d477eb8daba6006.32pc7fc8b4rwdvyl.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba600
           6.36f56o9lyq9gz5vm.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.3k1z8vrmtnl42i3r.rcgu.o" "/
           tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.3kmfcywxkph197hp.rcgu.o" "/tmp/user/spack-stage/spack-st
           age-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.3ocul9apjvjqx075.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7
           ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.3sf87l3ozb9uo64v.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src
           /build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.42czkdnwo7bfd82x.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/b
           ootstrap-8d477eb8daba6006.47o3eya703ccflyu.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.480
           k8fnn0kqsp52g.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.4iof8zti0usxcrp6.rcgu.o" "/tmp/a
           jwr/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.4mp51zk5swob8cu8.rcgu.o" "/tmp/user/spack-stage/spack-stage-r
           ust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.4oq4vsj88vfrizwd.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c
           4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.4pc6sp28wmcr6c58.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/buil
           d/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.53w36oxqhrizqvyz.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootst
           rap-8d477eb8daba6006.541598r83cx5dz0r.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.54tfs985
           xc8mfbl6.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.561i6mydc9hhp4sy.rcgu.o" "/tmp/user/s
           pack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.5cn08stbj7aka236.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1
           .73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.i1pbr8o636vx4s.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogx
           t662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.k1fd040p7x7d9ip.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootst
           rap/debug/deps/bootstrap-8d477eb8daba6006.wjotruzlf1ea0dx.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477
           eb8daba6006.yiks8f38btd5n0s.rcgu.o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006.3bd99vruv2s6czzo.r
           cgu.o" "-Wl,--as-needed" "-L" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps" "-L" "src/backend/linux_raw/arch/outline/debug" "-
           L" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/xz-5.4.1-itw2tpgrejzq22afnlx4wbv53c3j5eqo/lib" "-L" "/opt/spack_nov23/var/spack/environments/intel/instal
           l/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-lo
           iur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libfd_lock-20b02cc35d29f6ec.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/
           bootstrap/debug/deps/librustix-6c94f7e743aa6703.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libio_lifetimes-8a2dba74dc6
           80f8c.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/liblinux_raw_sys-4a9bcaf41614f995.rlib" "/tmp/user/spack-stage/spack-
           stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libbootstrap-6e776a4a3895cbca.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt6
           62geh/spack-src/build/bootstrap/debug/deps/libcmake-437c7db06d996a35.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libhex
           -287acb8baef005c9.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libtar-fd9a92e452460317.rlib" "/tmp/user/spack-stage/spac
           k-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libxattr-f351845ccfdf155a.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662
           geh/spack-src/build/bootstrap/debug/deps/libtoml-05e2dbdbe0c7257d.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libserde_
           json-07c7bb611aac4694.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libryu-4429e780fb4d0cf7.rlib" "/tmp/user/spack-stage/
           spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libitoa-4ff06280d8102809.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt
           662geh/spack-src/build/bootstrap/debug/deps/libcc-3cf9be60b14ebf88.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libopene
           r-1fbc67e13f71d144.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libclap_complete-08ca937ad0cac19b.rlib" "/tmp/user/spack
           -stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libtermcolor-25fd73655e1c9369.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7s
           sh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libignore-153b39bd4c1b5db7.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/deb
           ug/deps/libcrossbeam_utils-9401ffb533e8a1e6.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libwalkdir-6bfb14a7b9f17b77.rli
           b" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libsame_file-0bbe28eec5765210.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.
           73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libthread_local-6a2fb455a8ffc223.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spac
           k-src/build/bootstrap/debug/deps/libglobset-1f8721b5057115c3.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libfnv-b62b983
           5809230f7.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/liblog-3742ca8d465db47a.rlib" "/tmp/user/spack-stage/spack-stage-
           rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libregex-71ea01c2c4b52c89.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spac
           k-src/build/bootstrap/debug/deps/libregex_syntax-0dcec6a5ff950738.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libbstr-5
           2e41ceb3e9330b7.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/liblazy_static-66a7f92e5f9d7720.rlib" "/tmp/user/spack-stag
           e/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libregex_automata-2d84d6a14a727e2d.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7s
           sh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libaho_corasick-b7c754848b8c3820.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstr
           ap/debug/deps/libsha2-c340600a7f01d281.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libcpufeatures-6419cb802c64e361.rlib
           " "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libdigest-19b76e2e2fc97fec.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0
           -loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libblock_buffer-4878c514e219d6e8.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-sr
           c/build/bootstrap/debug/deps/libcrypto_common-1f2ccfa8d4e77883.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libgeneric_a
           rray-bb39a3aebe74be81.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libtypenum-f97854de311128c6.rlib" "/tmp/user/spack-st
           age/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libxz2-a6e79421cee94105.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iako
           gxt662geh/spack-src/build/bootstrap/debug/deps/liblzma_sys-4d8e0a3371dff4a4.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps
           /libobject-435a573070a878ba.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libmemchr-9ece6edae12fde31.rlib" "/tmp/user/spa
           ck-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libsemver-4ce2da0e3b2ea1a9.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ss
           h6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libclap-e9d2f0c316aa668e.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/
           deps/libclap_builder-c741e3a8e69349d4.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libanstyle-8002a47d829b5824.rlib" "/t
           mp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libclap_lex-81f704f1b8495692.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-lo
           iur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libbitflags-d001f5eedf12dd03.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build
           /bootstrap/debug/deps/libonce_cell-0bb103c89196941c.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libfiletime-5334460c066
           5bc3d.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/liblibc-2dd9de1036e23e99.rlib" "/tmp/user/spack-stage/spack-stage-rus
           t-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libcfg_if-5a036a9b8dd6f192.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-
           src/build/bootstrap/debug/deps/libbuild_helper-43e1bcbd2a5fdf0d.rlib" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/libserde-2c
           d60b50cabc4b93.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux
           -gnu/lib/libstd-6498d8891e016dca.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x8
           6_64-unknown-linux-gnu/lib/libpanic_unwind-3debdee1a9058d84.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2
           d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-8339c5bd5cbc92bf.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0
           -wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-160ebcebb54c11ba.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/
           rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-95c75789f1b65e37.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-cento
           s8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-7e8094f2d6258832.rlib" "/opt/spack_nov23/var/spack/environments/intel
           /install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-bac9783ef1b45db0.rlib" "/opt/spack_nov23
           /var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-a1cd87df2f2d8e7
           6.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libh
           ashbrown-7fd06d468d7dba16.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-un
           known-linux-gnu/lib/librustc_std_workspace_alloc-5ac19487656e05bf.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4a
           mbymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-c7c35d32cf825c11.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-boots
           trap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-c523f1571362e70b.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-
           2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-85f17c92b770a911.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linu
           x-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-598d3ba148dadcea.rlib" "/opt/spack_nov23/var/spack/environmen
           ts/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-a58ec2dab545caa4.rlib" "/opt/spack_nov23/v
           ar/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-f9dda8cca149f0fc.rlib"
            "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_st
           d_workspace_core-7ba4c315dd7a3503.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x
           86_64-unknown-linux-gnu/lib/libcore-5ac2993e19124966.rlib" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2d
           uvgb3/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-df2fb7f50dec519a.rlib" "-Wl,-Bdynamic" "-llzma" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl
           ,-z,noexecstack" "-L" "/opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/lib/rustlib/x86_64-unknown-linu
           x-gnu/lib" "-o" "/tmp/user/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/build/bootstrap/debug/deps/bootstrap-8d477eb8daba6006" "-Wl,--gc-sections" "-pie" "-Wl,-z,re
           lro,-z,now" "-nodefaultlibs"
     87      = note: icc: remark #10441: The Intel(R) C++ Compiler Classic (ICC) is deprecated and will be removed from product release in the second half of 2023. The Intel(R) oneAPI DPC++/C++ Compiler (ICX) 
           is the recommended compiler moving forward. Please transition to use this compiler. Use '-diag-disable=10441' to disable this message.
  >> 88              ld: /opt/intel/oneapi/compiler/2023.2.1/linux/compiler/lib/intel64_lin//libimf.a(libm_feature_flag.o): undefined reference to symbol '__intel_cpu_feature_indicator_x'
  >> 89              //opt/intel/oneapi/compiler/2023.2.1/linux/compiler/lib/intel64_lin//libintlc.so.5: error adding symbols: DSO missing from command line
     90    
     91      = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
     92      = note: use the `-l` flag to specify native libraries to link
     93      = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)
     94    
  >> 95    error: could not compile `bootstrap` (bin "bootstrap") due to previous error
     96    Building bootstrap
  >> 97    failed to run: /opt/spack_nov23/var/spack/environments/intel/install/linux-centos8-zen3/intel-2021.10.0/rust-bootstrap-1.73.0-wwn45x3ghjbu4ambymj2d6tpz2duvgb3/bin/cargo build --manifest-path /tmp/aj
           wr/spack-stage/spack-stage-rust-1.73.0-loiur4ftbc2h7ssh6c4iakogxt662geh/spack-src/src/bootstrap/Cargo.toml
     98    Build completed unsuccessfully in 0:00:13
```

### Information on your system

```console
* **Spack:** 0.22.0.dev0 (81e7d39bd2d263f425d8379e6420fe104af7768d)
* **Python:** 3.6.8
* **Platform:** linux-centos8-zen3
* **Concretizer:** clingo
```
 (same issue occurs on other systems, such as GitHub Actions ubuntu-latest runner)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

--- a/var/spack/repos/builtin/packages/rust/package.py
+++ b/var/spack/repos/builtin/packages/rust/package.py
@@ -98,6 +98,8 @@ class Rust(Package):
     conflicts("%gcc@:7.3", when="@1.73:", msg="Host GCC version must be at least 7.4")
     # https://github.com/rust-lang/llvm-project/commit/4d039a7a71899038b3bc6ed6fe5a8a48d915caa0
     conflicts("%gcc@13:", when="@:1.63", msg="Rust<1.64 not compatible with GCC>=13")
+    conflicts("%intel", msg="Rust not compatible with Intel Classic compilers")
+    conflicts("%oneapi", msg="Rust not compatible with Intel oneAPI compilers")
 
     extendable = True
     executables = ["^rustc$", "^cargo$"]

